docs/index.rst

```diff
@@ -431,6 +493,234 @@ A dialogue can be archived by setting ``is_archived`` to ``true`` when :ref:`rep
+    parameter, in the order the parameters were specified. Defaults to
+    ``-number``.
+
+Ordering revisions
+~~~~~~~~~~~~~~~~~~
+
++--------------+------------------------------------------------------------+
+| Parameter    | Description                                                |
++==============+============================================================+
+| ``number``   | Return revisions in ascending order of revision number     |
++--------------+------------------------------------------------------------+
+| ``-number``  | Return revisions in descending order of revision number    |
++--------------+------------------------------------------------------------+
+| ``created``  | Return revisions in ascending order of creation date       |
++--------------+------------------------------------------------------------+
+| ``-created`` | Return revisions in descending order of creation date      |
++--------------+------------------------------------------------------------+
+
+.. http:post:: /projects/(str:project_id)/dialogues/(str:dialogue_id)/revisions/
+
+Creates a new revision for dialogue ``dialogue_id`` under project
+``project_id`` using the :ref:`description <data-revisions>` given in
+the request body and returns the created revisions's description, along with
+the generated ``id`` field and ``url`` field for accessing the revision
+description.
+
+Creating a new revision updates the dialogue's description. Any new requests
+to :ref:`retrieve <dialogues-get>` the dialogue will return a dialogue
+:ref:`description <data-dialogues>` with the new revision applied.
+
+  .. sourcecode:: http
+
+     POST /projects/23/dialogues/21/revisions/ HTTP/1.1
+     Content-Type: application/json
+
+     {
+       "type": "edit",
+       "details": {
+         "id": "start",
+         "title": "Start of sequence",
+       },
+       "properties": {
+         "edit_type": "new_sequence",
+         "patch": [{
+           "op": "add",
+           "path": "/sequences",
+           "value": {
+             "id": "start",
+             "title": "Start of sequence",
+             "blocks": []
+           }
+         }]
+       }
+     }
+
+  .. sourcecode:: http
+
+     HTTP/1.1 201 Created
+     Content-Type: application/json
+
+     {
+       "id": "1",
+       "number": 1,
+       "user_id": "17",
+       "created": 1459943775033,
+       "type": "edit",
+       "details": {
+         "id": "start",
+         "title": "Start of sequence",
+       },
+       "properties": {
+         "edit_type": "new_sequence",
+         "patch": [{
+           "op": "add",
+           "path": "/sequences",
+           "value": {
+             "id": "start",
+             "title": "Start of sequence",
+             "blocks": []
+           }
+         }]
+       }
+     }
+
+It is also possible to create revisions in bulk by providing an array of revisions. In this case, an array of revision descriptions will be returned:
```

Comment:
We need this since the UI will be sending over revisions in batches (the plan is to send over revisions not yet synced with the API at some recurring interval). REST doesn't really cater for bulk operations, so this approach is a bit un-restful.

I read a [comment](http://roy.gbiv.com/untangled/2008/rest-apis-must-be-hypertext-driven#comment-743) Roy Fielding made where he said that a need for bulk API operations indicates that one has not designed enough resources. I'm not sure if that applies here - we need a batch operation here to minimize the number of requests the client will be making, and to make sure batches of revisions are created atomically. A resource we might be missing is a `RevisionBatch` or `RevisionSet` resource, but it doesn't seem right creating a new abstraction because of a technical reason.